### PR TITLE
Site wide search

### PR DESCRIPTION
**Pull request by mlt47**
---

**Pull Request Check List**
- [ ] Unit Tests & Regression Tests Added (Optional)
- [ ] Removed Unnecessary Logs/System.Outs/Comments/TODOs
- [ ] Added enough Logging to monitor expected behaviour change
- [ ] Security - Injection - everything run by an interpreter (SQL, OS...) is either validated or escaped
- [ ] Security - Data Exposure - PII is not stored or sent unencrypted
- [ ] Security - Access Control - Check authorisation on every new endpoint
- [ ] Security - New dependency - configured sensibly not relying on defaults
- [ ] Security - New dependency - Searched for any know vulnerabilities
- [ ] Security - New dependency - Signed up team to mailing list
- [ ] Security - New dependency - Added to dependency list
- [ ] DB schema changes - postgres-rutherford-create-script updated
- [ ] DB schema changes - upgrade script created matching create script
- [ ] Updated Release Procedure & Documentation (& Considered Implications to Previous Versions)
- [ ] Peer-Reviewed

### CODE REVIEW

File: src/main/java/uk/ac/cam/cl/dtg/isaac/api/Constants.java [hunk 1]
```diff
@@ -42,6 +42,8 @@
     public static final String TOPIC_SUMMARY_PAGE_TYPE = "isaacTopicSummaryPage";
     public static final String EVENT_TYPE = "isaacEventPage";
 
+    public static final String SEARCHABLE_TAG = "searchable";
```
Comment (jsharkey13) [inline src/main/java/uk/ac/cam/cl/dtg/isaac/api/Constants.java:45]: Could this be `search_result`? I still regret my choice of `nofilter`, but it's much too hard to change it now. I think `search_result` slightly better explains things.

File: src/main/java/uk/ac/cam/cl/dtg/segue/dao/content/GitContentManager.java [hunk 2]
```diff
@@ -248,20 +263,92 @@ public final Content getContentDOById(final String version, final String id, boo
...
+                    numberOfExpectedShouldMatches += 1;
+                }
+
+                contentQuery.setMinimumShouldMatch(numberOfExpectedShouldMatches);
+                matchQuery.should(contentQuery);
+            }
+        }
+
+        // Do not include any content with a nofilter tag
+        matchQuery.mustNot(new MustMatchInstruction(Constants.TAGS_FIELDNAME, HIDE_FROM_FILTER_TAG));
```
Comment (jsharkey13) [inline src/main/java/uk/ac/cam/cl/dtg/segue/dao/content/GitContentManager.java:342]: We'd sort of like staff to be able to find these using search, but getting the user information down to this level would be a pain. Perhaps an overriden method with another boolean flag for `includeHiddenContent`? I don't know how painful that would be.

File: src/main/java/uk/ac/cam/cl/dtg/segue/dao/content/GitContentManager.java.rej [hunk 3]
```diff
@@ -0,0 +1,21 @@
+diff a/src/main/java/uk/ac/cam/cl/dtg/segue/dao/content/GitContentManager.java b/src/main/java/uk/ac/cam/cl/dtg/segue/dao/content/GitContentManager.java	(rejected hunks)
```
Comment (jsharkey13) [inline src/main/java/uk/ac/cam/cl/dtg/segue/dao/content/GitContentManager.java.rej:1]: Erm, this shouldn't have been committed! Although be sure that whatever patch you tried to apply did get applied fully.

Comment (mlt47) [reply src/main/java/uk/ac/cam/cl/dtg/segue/dao/content/GitContentManager.java.rej:1]: Haha oops! I did do some unusual git switching and reverting between implementations. I'll remove this.

File: src/main/java/uk/ac/cam/cl/dtg/segue/dao/content/GitContentManager.java (hunk 2 above)
Comment (mlt47) [reply src/main/java/uk/ac/cam/cl/dtg/segue/dao/content/GitContentManager.java:342]: OK, that sounds easy enough to achieve

File: src/main/java/uk/ac/cam/cl/dtg/isaac/api/Constants.java (hunk 1 above)
Comment (mlt47) [reply src/main/java/uk/ac/cam/cl/dtg/isaac/api/Constants.java:45]: Sure

**Pull request merged**